packages/toolkit/src/actionCreatorInvariantMiddleware.ts

```diff
@@ -0,0 +1,34 @@
+import type { Middleware } from 'redux'
+import { isActionCreator as isRTKAction } from './createAction'
+
+export interface ActionCreatorInvariantMiddlewareOptions {
+  /**
+   * The function to identify whether a value is an action creator.
+   * The default checks for a function with a static type property and match method.
+   */
+  isActionCreator?: (action: unknown) => action is Function & { type?: unknown }
+}
+
+export function getMessage(type?: unknown) {
```

Comment:
:question: My only question here is whether this is going to get properly tree-shaken.  We switched from Terser to ESBuild for minification, and I was seeing that the other dev check middleware needed to have message functions like this _inside_ the middleware, and inside an `else` in order to shake properly.

Let me do a prod build of a test app and see what happens.

Reply:
Okay, looks good:

![image](https://user-images.githubusercontent.com/1128784/236573423-017bffd9-e089-4882-8a16-4bd906b7b8b0.png)